detectors/gcp/cloud-function.go

```diff
@@ -0,0 +1,100 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strings"
+
+	"cloud.google.com/go/compute/metadata"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+var (
+	errNotOnGoogleCloudFunction = errors.New("cannot detect environment variables from Google Cloud Function")
+)
+
+const (
+	gcpFunctionNameKey = "K_SERVICE"
+)
+
+//NewCloudFunction will return an implementation for gcp cloud function resource detector
+func NewCloudFunction() resource.Detector {
+	return &CloudFunction{
+		client: &gcpClientImpl{},
+	}
+}
+
+type gcpClient interface {
```

Comment:
I don't think you need to redefine this type and functions, since they are already defined in the cloud run implementation in this package.

Reply:
Thankyou @dashpole , 
Pushed code-changes, which will use implementation from cloud-run.
Kindly review.